### PR TITLE
Converted basic blocks

### LINKED ISSUE #82: Convert basic Block classes

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/main/java/robosky/uplands/block/UplanderBeaconBlock.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    @Override
+    protected void appendProperties(StateManager.Builder<Block, BlockState> builder) {
+        builder.add(SMOKING);
+    }
+
+    @Override
+    public ActionResult onUse(BlockState blockState, World world, BlockPos blockPos, PlayerEntity player, Hand hand, BlockHitResult result) {
+        if (player.world.dimension.getType() == DimensionType.OVERWORLD) {
```
Comment (i509VCB) [inline src/main/java/robosky/uplands/block/UplanderBeaconBlock.java]: `!world.isClient()` check first before doing anything else

File: src/main/java/robosky/uplands/block/UplandsGrassBlock.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+public class UplandsGrassBlock extends Block {
+
+    public UplandsGrassBlock() {
+        super(FabricBlockSettings.of(Material.ORGANIC).ticksRandomly().strength(0.6f, 0.6f).sounds(BlockSoundGroup.GRASS).build());
+    }
+    
+    @Override
+    public void scheduledTick(BlockState blockState, ServerWorld serverWorld, BlockPos blockPos, Random random) {
+        if (!serverWorld.isClient)
```
Comment (i509VCB) [inline src/main/java/robosky/uplands/block/UplandsGrassBlock.java]: You already have a server world. No need to check if it is not a client world

**Review by i509VCB — APPROVED** [review]
LGTM

Maybe let another person review before merging.

File: src/main/java/robosky/uplands/block/LodestoneBlock.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                if (neighbor.getBlock() == this)
+                    return neighbor.get(DISTANCE) + 1;
+                else if (this.isFullOpaque(neighbor, world, pos.offset(dir)))
+                    return 1;
+                else
+                    return 4;
+            }
+        }
+
+        return 4;
```
Comment (kvverti) [inline src/main/java/robosky/uplands/block/LodestoneBlock.java]: This should return the minimum of the value for each direction.

Comment (kvverti) [reply src/main/java/robosky/uplands/block/LodestoneBlock.java]: Also, brackets around the if statements

File: src/main/java/robosky/uplands/block/UplandsOreBlock.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                this.dropExperience(world, blockPos, xp);
+        }
+    }
+
+    private int getExperienceWhenMined(Random random) {
+        return MathHelper.nextInt(random, oreType.start, oreType.end);
+    }
+
+    // There has to be a better way at doing this?
+    private static Map<UplandsOreBlock.UplandsOreType, UplandsOreBlock> initMap() {
```
Comment (kvverti) [inline src/main/java/robosky/uplands/block/UplandsOreBlock.java]: `static {}` block

File: src/main/java/robosky/uplands/clientmixins/FluidRendererMixin.java [hunk 5]
```diff
@@ -21,7 +23,7 @@
 import robosky.uplands.block.UplandsWaterBlock;
 import robosky.uplands.world.WorldRegistry;
 
-@Mixin(FluidRenderer.class)
+@Mixin(FluidRenderer.class) @Environment(EnvType.CLIENT)
```
Comment (kvverti) [inline src/main/java/robosky/uplands/clientmixins/FluidRendererMixin.java]: Separate line annotations

**Review by kvverti — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
Should be okay now

File: src/main/java/robosky/uplands/block/LodestoneBlock.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+                else if (this.isFullOpaque(neighbor, world, pos.offset(dir))) {
+                    return 1;
+                }
+                else {
+                    return 4;
+                }
+            }
+        }
+
+        return 0;
```
Comment (kvverti) [inline src/main/java/robosky/uplands/block/LodestoneBlock.java]: You're supposed to iterate through the entire loop, and take the minimum of the calculation across all six directions.

**Review by kvverti — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
I meant distance property

**Comment by kvverti** [comment]
The distance property determines how far away the lodestone block is from support (support is defined as the closest solid non-lodestone block, or the block underneath). If a lodestone block is more than four blocks away from support, it falls.

You can revert this commit with
`git revert`

**Review by kvverti — APPROVED** [review]

**Pull request merged**